packages/cadl-python/src/emitter.ts

```diff
@@ -355,7 +355,11 @@ function emitParameter(
         paramMap.implementation = "Client";
         paramMap.in_docstring = false;
     }
-    return { clientDefaultValue, ...base, ...paramMap };
+    if (parameter.type === "endpointPath") {
+        return { clientDefaultValue, ...base, ...paramMap, skipUrlEncoding: true };
+    } else {
+        return { clientDefaultValue, ...base, ...paramMap };
+    }
```

Comment:
@iscai-msft replied https://github.com/Azure/autorest.python/pull/1568#discussion_r1019508000: I understand that you mean that we want to skip url encode for `endpointPath` parameter, right?

Reply:
yeah, for all service endpoint path parameters, just don't focus on the name being "endpoint"

Reply:
@iscai-msft How about current logic? I judge whether its type is `endpintPath` to set `skipUrlEncoding`

Reply:
@msyyc I'm going to go ahead and split this pr into two. For the api versions, we have new helper methods in cadl-dpg that we should leverage here. And yes the current logic looks good, thanks! I'll create another PR to unblock @catalinaperalta for endpoint path, thank you for your work and sorry I have to cut in